fhir-model/src/main/java/com/ibm/fhir/model/util/ModelSupport.java

```diff
@@ -449,18 +1103,50 @@ public boolean hasBinding() {
         return Collections.unmodifiableMap(resourceTypeMap);
     }
 
-    private static List<Field> getAllFields(Class<?> modelClass) {
-        List<Field> fields = new ArrayList<>();
+    private static Map<String, ElementInfo> getElementInfoMap(Class<?> modelClass,
```

Comment:
Could use some comments to explain what's going on here

Reply:
sure.  most of this code already existed and is doing the same thing...my big addition was to use the elementInfoMapCache so that we're not calling 'getDeclaredFields' over and over on common superclasses like Element, BackboneElement, Resource, and DomainResource.

Reply:
added some comments...please see if it helps